hardware/detect_utils.py

```diff
@@ -231,3 +232,42 @@ def clean_str(val):
 def clean_tuples(lst):
     """Clean a list of tuples from bad UTF-8 strings."""
     return [tuple([clean_str(val) for val in elt]) for elt in lst]
+
+
+def _get_uuid_x86_64():
+    """Get uuid from dmidecode"""
+
+    uuid_cmd = subprocess.Popen(
+        "dmidecode -t 1 | grep UUID | awk '{print $2}'", shell=True,
```

Comment:
Can you take the opportunity to switch to "dmidecode -s system-uuid" ?

Reply:
I'd rather do this in a different patch, I prefer keeping refactoring work and logic changes separated

Reply:
ack

Reply:
Can you rebase and I'll merge it.

Reply:
will do, thanks!